File: test-clients/java-okhttp-moshi/src/test/java/test_client/CheckClientTest.java
package test_client;

import org.junit.jupiter.api.Test;
import test_client.clients.check.CheckClient;
import test_client.errors.*;
import test_client.errors.models.*;
import test_client.models.*;

import static org.junit.jupiter.api.Assertions.*;

public class CheckClientTest {
	private final String baseUrl = "http://localhost:8081";
	private final CheckClient client = new CheckClient(baseUrl);

	private final Message message = new Message(123, "the value");

	@Test
	public void checkEmpty_doesntThrowException() {
		assertDoesNotThrow(client::checkEmpty);
	}

	@Test
	public void checkEmptyResponse_doesntThrowException() {
		assertDoesNotThrow(() -> client.checkEmptyResponse(message));
	}

	@Test
	public void checkForbidden_doesntThrowException() {
		var exception = assertThrows(ClientException.class, client::checkForbidden);
		assertTrue(exception.getCause() instanceof ForbiddenException);
	}

	@Test
	public void checkConflict_checkExceptionBody() {
		var exception = (ConflictException) assertThrows(ClientException.class, client::checkConflict).getCause();
		var expectedBody = new ConflictMessage("Conflict with the current state of the target resource");
		assertEquals(exception.getBody(), expectedBody);
	}

	@Test
	public void checkConflict_throwException() {
		var exception = assertThrows(ClientException.class, client::checkConflict);
		assertTrue(exception.getCause() instanceof ConflictException);
	}

	@Test
	public void checkBadRequest_checkExceptionBody() {
		var exception = (BadRequestException) assertThrows(ClientException.class, client::checkBadRequest).getCause();
		var expectedBody = new BadRequestError("Failed to execute request", ErrorLocation.UNKNOWN, null);
		assertEquals(exception.getBody(), expectedBody);
	}

	@Test
	public void checkBadRequest_throwException() {
		var exception = assertThrows(ClientException.class, client::checkBadRequest);
		assertTrue(exception.getCause() instanceof BadRequestException);
	}
}

File: test-clients/kotlin/template/src/test/kotlin/theclient/CheckClientTest.kt
package theclient

import org.junit.jupiter.api.*
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import theclient.clients.check.CheckClient
import theclient.errors.*
import theclient.errors.models.*
import theclient.models.*

class CheckClientTest {
    private val client = CheckClient("http://localhost:8081")

    private val bodyMessage = Message(123, "the string")

    @Test
    fun checkEmpty_doesntThrowException() {
        assertDoesNotThrow { client.checkEmpty() }
    }

    @Test
    fun checkEmptyResponse_doesntThrowException() {
        assertDoesNotThrow { client.checkEmptyResponse(bodyMessage) }
    }

    @Test
    fun checkForbidden_throwException() {
        val exception = assertThrows<ClientException> { client.checkForbidden() }
        Assertions.assertTrue(exception.cause is ForbiddenException)
    }

    @Test
    fun checkConflict_checkExceptionBody() {
        val exception = assertThrows<ClientException> { client.checkConflict() }.cause as ConflictException
        val expectedBody = ConflictMessage("Conflict with the current state of the target resource")
        Assertions.assertEquals(exception.body, expectedBody)
    }

    @Test
    fun checkConflict_throwException() {
        val exception = assertThrows<ClientException> { client.checkConflict() }
        Assertions.assertTrue(exception.cause is ConflictException)
    }

    @Test
    fun checkBadRequest_checkExceptionBody() {
        val exception = assertThrows<ClientException> { client.checkBadRequest() }.cause as BadRequestException
        val expectedBody = BadRequestError("Failed to execute request", ErrorLocation.UNKNOWN, null)
        Assertions.assertEquals(exception.body, expectedBody)
    }

    @Test
    fun checkBadRequest_throwException() {
        val exception = assertThrows<ClientException> { client.checkBadRequest() }
        Assertions.assertTrue(exception.cause is BadRequestException)
    }
}
